src/Tribe/Cache.php

```diff
@@ -289,32 +289,21 @@ public function get_id( $key, $expiration_trigger = '' ) {
 	 * Returns the time of an action last occurrence.
 	 *
 	 * @since 4.9.14 Changed the return value type from `int` to `float`.
+	 * @since TBD No longer memoizes the first triggered timestamp.
 	 *
 	 * @param string $action The action to return the time for.
 	 *
 	 * @return float The time (microtime) an action last occurred, or the current microtime if it never occurred.
 	 */
 	public function get_last_occurrence( $action ) {
-		static $cache_var_name = __METHOD__;
-
-		$cache_last_actions = tribe_get_var( $cache_var_name, [] );
-
-		if ( isset( $cache_last_actions[ $action ] ) ) {
-			return $cache_last_actions[ $action ];
-		}
-
 		$last_action = (float) get_option( 'tribe_last_' . $action, null );
```

Comment:
We need to make sure if the result of this, when `get_option( KEY, '__UNIQUE_DEFAULT_KEY__' ) === '__UNIQUE_DEFAULT_KEY__'` we do a `update_option( KEY, null, true );` so that the value is auto-loaded next time.

Reply:
So we want it to save/return `null` if not saved yet?

Reply:
To clarify on my question above - right now it returns the `microtime()` value. If I understand you correctly, you are asking to change it to return `null` if it's not set instead of the current `microtime()` ? That is a change in logic (which is why I left things as they are).

And regarding the auto-load - the logic as it is should already be doing that.

Reply:
No no. What i am trying to suggest is that when you have no value in the DB, WordPress wont preload any value thus creating a database request to look for the value, what I am asking you to make sure that once we verify that the value is not existent just fill with something that will be preloaded next time, as a way to avoid this query being present every time in case the option doesn't exist.

Basically preventing a bunch of calls to `tribe_last_{$action}` to exist more than in one request. Afterwards there will always be a preloaded value even if `null`.

Reply:
Isn't that what this is doing already?

https://github.com/the-events-calendar/tribe-common/pull/1909/files#diff-03cdddb06b33fcad3e0a4ae404379bf6fcdb2eb19c5cfbc1444c55001d2ddec5L308-L310